File: src/commands/install-prompts.js
"use strict";

const fs = require("fs");
const path = require("path");
const {
  getPromptsDir,
  getIdePromptsTarget,
  readConfig,
  updateGitignore,
} = require("../config");

async function installPromptsCommand(cwd) {
  const { checkbox } = require("@inquirer/prompts");
  cwd = cwd || process.cwd();
  const config = readConfig(cwd);

  if (!config) {
    console.log('No wizard configuration found. Run "wizard install" first.');
    return;
  }

  const promptsDir = getPromptsDir();
  const promptFiles = fs
    .readdirSync(promptsDir)
    .filter((f) => f.endsWith(".md"));

  if (promptFiles.length === 0) {
    console.log("No prompt templates available.");
    return;
  }

  const choices = promptFiles
    .sort()
    .map((f) => ({ name: f.replace(".md", ""), value: f }));

  const selected = await checkbox({
    message: "Select prompts to install:",
    choices,
  });

  if (selected.length === 0) {
    console.log("No prompts selected. Aborting.");
    return;
  }

  const targets = getIdePromptsTarget(cwd, config.ides);

  for (const [ide, targetDir] of Object.entries(targets)) {
    fs.mkdirSync(targetDir, { recursive: true });
    for (const prompt of selected) {
      const src = path.join(promptsDir, prompt);
      const dest = path.join(targetDir, prompt);
      fs.copyFileSync(src, dest);
    }
    const relDir = path.relative(cwd, targetDir);
    updateGitignore(cwd, [relDir + "/"]);
    console.log(`Prompts installed to ${relDir}/`);
  }
}

module.exports = { installPromptsCommand };

File: wizard/commands/install_prompts.py
"""Install prompts command."""

import os
import shutil

from InquirerPy import inquirer

from wizard.config import get_ide_prompts_target, get_prompts_dir, read_config


def install_prompts_command(cwd: str | None = None) -> None:
    """Run the install prompts command."""
    cwd = cwd or os.getcwd()
    config = read_config(cwd)

    if not config:
        print('No wizard configuration found. Run "wizard install" first.')
        return

    prompts_dir = get_prompts_dir()
    prompt_files = [f for f in os.listdir(prompts_dir) if f.endswith(".md")]

    if not prompt_files:
        print("No prompt templates available.")
        return

    choices = [{"name": f.replace(".md", ""), "value": f} for f in sorted(prompt_files)]

    selected = inquirer.checkbox(
        message="Select prompts to install:",
        choices=choices,
    ).execute()

    if not selected:
        print("No prompts selected. Aborting.")
        return

    targets = get_ide_prompts_target(cwd, config["ides"])

    for ide, target_dir in targets.items():
        os.makedirs(target_dir, exist_ok=True)
        for prompt in selected:
            src = os.path.join(prompts_dir, prompt)
            dest = os.path.join(target_dir, prompt)
            shutil.copy2(src, dest)
        print(f"Prompts installed to {os.path.relpath(target_dir, cwd)}/")
